### PR TITLE
[1812] Setup course delete

### PR DESCRIPTION
**Pull request by dankmitchell**
### Context
Deleting courses

### Changes proposed in this pull request
Add real course delete functionality

### Guidance to review
`/organisations/:provider_code/2019/courses/:course_code/delete`

# After
localhost_3000_organisations_1P2_2019_courses_2RH7_delete(Laptop with MDPI screen)

### CODE REVIEW

File: app/views/courses/delete.html.erb [hunk 1]
```diff
@@ -21,14 +21,19 @@
...
-    <p class="govuk-body">In your email include:</p>
-    <ul class="govuk-list govuk-list--bullet">
-      <li>course code</li>
-      <li>course title</li>
-    </ul>
+    <h2 class="govuk-heading-m">Confirm delete</h2>
+
+    <%= form_with model: @course, url: provider_recruitment_cycle_course_path(@provider.provider_code, course.recruitment_cycle_year, course.course_code), method: :delete do |f| %>
+      <div class="govuk-form-group">
+        <%= f.label 'confirm_course_code', 'Type in the course code to confirm', class: 'govuk-label govuk-label--s' %>
```
Comment (fofr) [inline app/views/courses/delete.html.erb]: We don't need the `govuk-label--s` here

File: app/controllers/courses_controller.rb [hunk 2]
```diff
@@ -99,6 +99,17 @@ def delete; end
 
   def preview; end
 
+  def destroy
+    if params[:course][:confirm_course_code] == @course.course_code
+      @course.destroy
+      redirect_to provider_recruitment_cycle_courses_path(@provider.provider_code, @provider.recruitment_cycle_year)
+      flash[:success] = "Course deleted"
```
Comment (fofr) [inline app/controllers/courses_controller.rb]:
```suggestion
      flash[:success] = "#{@course.name} (#{@course.course_code}) has been deleted"
```

**Comment by fofr** [comment]
Looking good :100:

File: app/controllers/courses_controller.rb (hunk 2 above)
Comment (dankmitchell) [reply app/controllers/courses_controller.rb]: Nice. I copied the prototype ;-)

File: config/routes.rb [hunk 3]
```diff
@@ -49,6 +49,7 @@
 
         get '/withdraw', on: :member, to: 'courses#withdraw'
         get '/delete', on: :member, to: 'courses#delete'
+        delete '/:id', on: :member, to: 'courses#destroy'
```
Comment (defong) [inline config/routes.rb]: should it be id?

Comment (dankmitchell) [reply config/routes.rb]: Removed :)

**Comment by fofr** [comment]
[[M] Delete a course (MCFE)](https://trello.com/c/wnSSuqeO/1812-m-delete-a-course-mcfe)

**Comment by dankmitchell** [comment]
Can I merge this now @timabell 🤓 ?

**Pull request merged**